### PR TITLE
documentation for dependency_parsing.py

### PR DESCRIPTION
**Pull request by varmichelle**

### CODE REVIEW

File: gender_novels/analysis/dependency_parsing.py [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import urllib
 from nltk.parse.stanford import StanfordDependencyParser
```
Comment (srisi) [inline gender_novels/analysis/dependency_parsing.py]: make sure to update the dependency parser to the non-deprecated one:
`DeprecationWarning: The StanfordDependencyParser will be deprecated. Please use nltk.parse.corenlp.StanforCoreNLPDependencyParser instead`

File: gender_novels/analysis/dependency_parsing.py [hunk 2]
```diff
@@ -15,6 +19,69 @@ def load_jars(path_to_jar, path_to_models_jar):
...
+    return (male_subj_count, male_obj_count, female_subj_count, female_obj_count)
+
+
+def parse_sentence(sentence):
+    """
+    This function does all sentence parsing (we cannot split this up into separate functions for
+    performance reasons (each additional function will require iterating over the entire tree again)
+    :param sentence: sentence (string) we want to parse
+    :return: the counts of male and female subject and object occurrences as a tuple of 4
+    """
```
Comment (srisi) [inline gender_novels/analysis/dependency_parsing.py]: Here is a sample of how a doc test might look like. You may need to pass the parser as an argument into the function.
```
    >>> path_to_jar = "assets/stanford-parser.jar"
    >>> path_to_models_jar = "assets/stanford-parser-3.9.1-models.jar"
    >>> load_jars(path_to_jar, path_to_models_jar)
    >>> dependency_parser = StanfordDependencyParser(path_to_jar, path_to_models_jar)
    >>> sentence = "Here is a test sentence that we want to parsep"
    >>> parse_sentence(sentence)
    Dependency Tree of the Sentence
```

File: gender_novels/analysis/dependency_parsing.py [hunk 3]
```diff
@@ -15,6 +19,69 @@ def load_jars(path_to_jar, path_to_models_jar):
...
+        t_female_obj_count += female_obj_count
+        # print(sentence)
+        # print cumulative counts
+        print(t_male_subj_count, t_male_obj_count, t_female_subj_count, t_female_obj_count)
+    return (t_male_subj_count, t_male_obj_count, t_female_subj_count, t_female_obj_count)
+
+
 if __name__ == "__main__":
 
     # create dependency parser
```
Comment (srisi) [inline gender_novels/analysis/dependency_parsing.py]: everything in the __name__ == '__main__' section should be in a function of its own, for example test_analysis(). That will make it easy to later on write multiple analyses of a whole novel or corpus.

File: gender_novels/analysis/dependency_parsing.py (hunk 1 above)
Comment (varmichelle) [reply gender_novels/analysis/dependency_parsing.py]: I looked into this a while back and ended up getting it to work, but it requires setting up a server  locally in order to communicate with the parser via REST API. It seemed too complicated so I stuck with the deprecated version.

**Comment by varmichelle** [comment]
Thanks for the review. I've added doctests for all the functions with return statements, but I'm unsure of how to write the test for parse_novel since its return statement is the cumulative pronoun frequency. My other doctests don't seem to be running properly; I'm not getting errors if I change the output to something incorrect. Do I need to run them in a special way?

**Comment by srisi** [comment]
> Thanks for the review. I've added doctests for all the functions with return statements, but I'm unsure of how to write the test for parse_novel since its return statement is the cumulative pronoun frequency. My other doctests don't seem to be running properly; I'm not getting errors if I change the output to something incorrect. Do I need to run them in a special way?

Thanks. The tests are very helpful. For the parse_novel function, it would be best if you could actually parse a whole novel but your current solution works fine. 
My issue with the function is that when I try to parse a whole novel, I get a java error. 
Sample code: 
```
    >>> parser = get_parser("assets/stanford-parser.jar","assets/stanford-parser-3.9.1-models.jar")
    >>> novel_metadata = {'author': 'Hawthorne, Nathaniel', 'title': 'Scarlet Letter',
    ...                   'corpus_name': 'sample_novels', 'date': '1900',
    ...                   'filename': 'hawthorne_scarlet.txt'}
    >>> toy_novel = Novel(novel_metadata)
    >>> parse_novel(toy_novel, parser)
```
This gives me the following error:
`    OSError: Java command failed : ['/usr/local/java/jdk1.8.0_45/bin/java', '-mx1000m', '-cp', 'assets/stanford-parser-3.9.1-models.jar:assets/stanford-parser-3.9.1-models.jar:assets/stanford-parser.jar', 'edu.stanford.nlp.parser.lexparser.LexicalizedParser', '-model', 'edu/stanford/nlp/models/lexparser/englishPCFG.ser.gz', '-sentences', 'newline', '-outputFormat', 'conll2007', '-encoding', 'utf8', '/tmp/tmpi1tcer_p']`
Any idea or advise on what's happening here?

**Comment by varmichelle** [comment]
Yeah I was able to get the doctests to work later. I agree for the parse_novel doctest, but it takes ~1 sec per sentence to analyze, so an entire novel would take too much time. I'm not sure what's going on with parse_novel; I've never seen that error before. The code works on my machine and Derek's though.

**Comment by srisi** [comment]
Thanks!

**Pull request merged**